packages/dashboard/src/components/sidePanel/sections/baseSettingSection/index.tsx

```diff
@@ -89,28 +106,47 @@ export const BaseSettings: FC<Widget> = (widget) => {
         />
 
         <label htmlFor='base-settings-width' className='section-item-label'>
-          {defaultMessages.width}
+          Width
         </label>
-        <Input
-          controlId='base-settings-width'
-          value={`${formattedValue.width}`}
-          type='number'
-          onChange={onWidthChange}
-          data-test-id='base-setting-width-input'
+
+        <Controller
+          name='width'
+          control={control}
+          render={({ field: { value } }) => (
+            <Input
+              controlId='base-settings-width'
+              data-test-id='base-setting-width-input'
+              invalid={Boolean(errors.width)}
+              onBlur={() => trigger('width')}
+              onChange={onWidthChange}
+              type='number'
+              value={String(value)}
+            />
+          )}
+          rules={{ required: true, min: 2, max: 100 }}
         />
 
         <label htmlFor='base-settings-height' className='section-item-label'>
-          {defaultMessages.height}
+          Height
         </label>
-        <Input
-          controlId='base-settings-height'
-          value={`${formattedValue.height}`}
-          type='number'
-          onChange={onHeightChange}
-          data-test-id='base-setting-height-input'
+
+        <Controller
+          name='height'
+          control={control}
+          render={({ field: { value } }) => (
+            <Input
+              controlId='base-settings-height'
+              data-test-id='base-setting-height-input'
+              invalid={Boolean(errors.height)}
```

Comment:
Blocker: Do not manually set invalid state. Use [`fieldState.error`](https://react-hook-form.com/api/usecontroller/controller/) combined with [`<FormField error />`](https://cloudscape.design/components/form-field/?tabId=playground&example=with-error) to render invalid state.